app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+  validates :email, uniqueness: true
+  has_secure_password
+  private
+  def downcase_email
```

Comment:
```suggestion
  before_save :downcase_email
  
  validates :email, uniqueness: true
  
  has_secure_password
  
  private
  
  def downcase_email
```

ngoài ra, giải thích code `has_secure_password` nghe coi

Reply:
dạ cái has_secure_password sử dụng bcrypt để băm mật khẩu, kiểu lưu mật khẩu dưới dạng hash chứ không phải mật khẩu thực hiển thị của người dùng á chị